R/pairwise-comparisons.R

```diff
@@ -1,24 +1,53 @@
 #' combinations of two models are compared against each other based on the
 #' overlapping set of available forecasts common to both models.
-#' Internally, a ratio of the mean scores of both models is computed.
-#' The relative score of a model is then the geometric mean of all mean score
-#' ratios which involve that model. When a baseline is provided, then that
-#' baseline is excluded from the relative scores for individual models
-#' (which therefore differ slightly from relative scores without a baseline)
-#' and all relative scores are scaled by (i.e. divided by) the relative score of
-#' the baseline model.
-#' Usually, the function input should be unsummarised scores as
-#' produced by [score()].
-#' Note that the function internally infers the *unit of a single forecast* by
-#' determining all columns in the input that do not correspond to metrics
-#' computed by [score()]. Adding unrelated columns will change results in an
-#' unpredictable way.
+#'
+#' The input should be a `scores` object as produced by [score()]. Note that
+#' adding additional unrelated columns can unpredictably change results.
+#' The output of the pairwise comparisons is a set of mean score ratios,
+#' relative skill scores and p-values.
+#'
+#' The following illustrates the pairwise comparison process:
+#'
+#' \if{html}{
+#'   \out{<div style="text-align: left">}
+#'   \figure{pairwise-illustration.png}{options: style="width:750px;max-width:100\%;"}
+#'   \out{</div>}
+#' }
+#' \if{latex}{
+#'   \figure{pairwise-illustration.png}
+#' }
+#'
+#' *Mean score ratios*
+#'
+#' For every pair of two models, a mean score ratio is computed. This is simply
+#' the mean score of the first model divided by the mean score of the second.
+#' Mean score ratios are computed based on the set of overlapping forecasts
+#' between the two models. That means that only scores for those targets are
+#' taken into account for which both models have submitted a forecast.
+#'
+#' *(Scaled) Relative skill scores*
+#'
+#' The relative score of a model is the geometric mean of all mean score
+#' ratios which involve that model.
+#' If a baseline is provided, scaled relative skill scores will be calculated
+#' as well. Scaled relative skill scores are simply the relative skill score of
+#' a model divided by the relative skill score of the baseline model.
+#'
+#' *p-values*
+#'
+#' In addition, the function computes p-values for the comparison between two
+#' models (again based on the set of overlapping forecasts). P-values can be
+#' computed in two ways: based on a nonparametric Wilcoxon signed-rank test
+#' (internally using [wilcox.test()] with `paired = TRUE`) or based on a
+#' permutation test. The permutation test is based on the difference in mean
+#' scores between two models. The null hypothesis is that the mean score
+#' difference is zero (see [permutation_test()]).
+#' Adjusted p-values are computed by calling [p.adjust()] on the raw p-values.
```

Comment:
```suggestion
#' (internally using [stats::wilcox.test()] with `paired = TRUE`) with a null hypothesis that the ratio of scores of the two models is symmetric around 1 or, alternatively, based on a
#' permutation test of the difference in mean
#' scores between two models, with a null hypothesis is that the mean score
#' difference is zero (see [permutation_test()]).
#' The method can be selected using the `test_type` argument, which is passed to [compare_two_models()].
#' 
#' Adjusted p-values are computed by calling [stats::p.adjust()] on the raw p-values.
```

"null hypothesis is that the mean score difference is zero"
Can we just replace that by mean ratio being 1? If yes then that would link better with the score we're actually calculating. 

Also given the prominence of the test type in the documentation perhaps make `test_type` an explicit argument rather than passing it via ellipsis?

Reply:
`permutation_test()` has an option to choose either difference or mean ratio == 1 🙈 The default is mean difference (although I have to admit I don't have a good intuition for how much they would differ in practice or whether they should differ at all. If I remember correctly, there was something related to whether you have a one-sided or a two-sided p-value and in some cases both p-values coincide. I remember some conversations with Johannes about this.

Reply:
Would you be ok with leaving this for a separate issue?

Reply:
Made a new issue, #769.